snuba/state/__init__.py

```diff
@@ -104,59 +114,46 @@ def numeric(value: Any) -> Any:
             return value
 
 
-ABTEST_RE = re.compile("(?:(-?\d+\.?\d*)(?:\:(\d+))?\/?)")
-
-
-def abtest(value: Optional[Any]) -> Optional[Any]:
-    """
-    Recognizes a value that consists of a '/'-separated sequence of
-    value:weight tuples. Value is numeric. Weight is an optional integer and
-    defaults to 1. Returns a weighted random value from the set of values.
-    eg.
-    1000/2000 => returns 1000 or 2000 with equal weight
-    1000:1/2000:1 => returns 1000 or 2000 with equal weight
-    1000:2/2000:1 => returns 1000 twice as often as 2000
-    """
-    if isinstance(value, str) and ABTEST_RE.match(value):
-        values = ABTEST_RE.findall(value)
-        total_weight = sum(int(weight or 1) for (_, weight) in values)
-        r = random.randint(1, total_weight)
-        i = 0
-        for (v, weight) in values:
-            i += int(weight or 1)
-            if i >= r:
-                return numeric(v)
-
-    return value
-
-
-def set_config(key: str, value: Optional[Any], user: Optional[str] = None) -> None:
-    if value is not None:
-        value = "{}".format(value).encode("utf-8")
+def set_config(
+    key: str, value: Optional[Any], user: Optional[str] = None, force: bool = False
+) -> None:
+    value = get_typed_value(value)
+    enc_value = "{}".format(value).encode("utf-8") if value is not None else None
 
     try:
-        original_value = rds.hget(config_hash, key)
-        if value == original_value:
-            return
+        enc_original_value = rds.hget(config_hash, key)
+        if enc_original_value is not None and value is not None:
+            original_value = get_typed_value(enc_original_value.decode("utf-8"))
+            if value == original_value:
+                return
```

Comment:
Is this intentional ?
```
>>> a = 1
>>> type(a)
<class 'int'>
>>> b = 1.0
>>> type(b)
<class 'float'>
>>> a == b
True
```

Basically an int can be equal to a float if they represent the same number but in terms of types they would still be different.

Reply:
Added a check for the type on this line as well